protocol/rpcprovider/rpcprovider_server.go

```diff
@@ -185,10 +185,10 @@ func (rpcps *RPCProviderServer) craftChainMessage() (chainMessage chainlib.Chain
 func (rpcps *RPCProviderServer) Relay(ctx context.Context, request *pairingtypes.RelayRequest) (*pairingtypes.RelayReply, error) {
 	// count the number of simultaneous relay calls
 	rpcps.providerLoadManager.addRelayCall()
-	defer rpcps.providerLoadManager.removeRelayCall()
-	provider_relay_load := strconv.FormatFloat(rpcps.providerLoadManager.getProviderLoad(), 'f', -1, 64)
-	trailer_md := metadata.Pairs(chainlib.RpcProviderUniqueIdHeader, rpcps.providerUniqueId, chainlib.RpcProviderLoadRateHeader, provider_relay_load)
-	grpc.SetTrailer(ctx, trailer_md)
+	defer func() { go rpcps.providerLoadManager.subtractRelayCall() }()
+	provideRelayLoad := rpcps.providerLoadManager.getProviderLoad()
```

Comment:
Last review I wanted the provider load manager to create a meta data pair and just return it (or an empty array, 

a better solution would be just passing the ctx, and setting the trailer inside. you can have a multiple set trailer functions called they do not overwrite each other they just append. so entire flow for appending the meta data to the context's trailer should happen inside the providerLoadManager methods. 

you can call it applyProviderLoadMetadataToContextTrailer

Reply:
Fixed in [9eabb5a](https://github.com/lavanet/lava/pull/1720/commits/9eabb5ac988c7c12d533a62a1f2bfa88eeb9425a)